src/pages/crm-cleint-about/ChangeRole.tsx

```diff
@@ -0,0 +1,108 @@
+import styles from './crmClientAbout.module.scss'
+
+import * as React from 'react'
+import Button from '@mui/material/Button'
+import Menu from '@mui/material/Menu'
+import MenuItem from '@mui/material/MenuItem'
+import { Box, Typography } from '@mui/material'
+import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown'
+import DoneIcon from '@mui/icons-material/Done'
+
+import { ROLE_TRANSLATIONS } from '../crm-clients-page/CrmTableStickyHead'
+import { Role, UserResponseForCRM } from '../../types'
+import axiosInstance from '../../axios/settings'
+import ConfirmModal from './ConfirmModal'
+import { changeRoleStyle, roleList, saveNewRole } from './style'
+
+const ROLES = ['admin', 'moderator', 'user']
+
+type BasicMenuProps = {
+  user: UserResponseForCRM
+  userRole: string
+  setUserRole: (role: Role) => void
+}
+
+export default function BasicMenu({
+  user,
+  userRole,
+  setUserRole
+}: BasicMenuProps) {
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const [selectedRole, setSelectedRole] = React.useState<string>(user.role)
+  const [openModal, setOpenModal] = React.useState<boolean>(false)
+
+  const open = Boolean(anchorEl)
+
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) =>
+    setAnchorEl(event.currentTarget)
+
+  const handleClose = () => setAnchorEl(null)
+
+  const handleMenuItemClick = (role: string) => setSelectedRole(role)
+
+  const fetchChangeRole = async (role: string) => {
+    try {
+      const { data } = await axiosInstance.put(`/api/users/change_role`, {
+        id: user.id,
+        role,
+        updated_at: new Date().toISOString()
+      })
+      setUserRole(data.role)
+      setOpenModal(false)
+    } catch (error) {
+      console.error('Помилка зміни ролі:', error)
+    }
+  }
+
+  const changeRole = () => fetchChangeRole(selectedRole)
```

Comment:
why do you need this function, use just `fetchChangeRole(selectedRole)`

Reply:
Done